primaryHandler.go

```diff
@@ -68,12 +68,21 @@ func GetLogger(ctx context.Context) bascule.Logger {
 	return logger
 }
 
-func populateMessage(ctx context.Context, message *wrp.Message, logger log.Logger) {
+func ensureWRPMessageIntegrity(ctx context.Context, m *wrp.Message, violatorsCount metrics.Counter) {
 	if auth, ok := bascule.FromContext(ctx); ok {
-		if token := auth.Token; token != nil {
-			var claims claims
-			mapstructure.Decode(token.Attributes(), &claims)
-			message.PartnerIDs = claims.AllowedResources.AllowedPartners
+		var (
+			token       = auth.Token
+			satClientID = "none"
+			attributes  = token.Attributes()
+		)
+
+		if principal := token.Principal(); len(principal) > 0 {
+			satClientID = principal
+		}
+
+		if len(m.PartnerIDs) < 1 {
+			violatorsCount.With(ClientIDLabel, satClientID).Add(1)
+			m.PartnerIDs, _ = attributes.GetStringSlice(PartnerIDClaimsKey)
```

Comment:
I think we should make sure getting the partner ids here was okay.

Reply:
Yeah, I think these are two good points! I'll make the changes 😄